File: BOJ/[2026] 소풍/woong-jae/2026.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

const input = [];
rl.on('line', function (line) {
  input.push(line.split(' '));
}).on("close", function () {
  console.log(solution());
  process.exit();
});

function solution() {
  const [K, N] = input[0].map(e => +e);
  const friends = input.slice(1);

  const adjGraph = Array.from(Array(N + 1), () => Array(N + 1).fill(false));
  friends.forEach(([from, to]) => {
    adjGraph[+from][+to] = true;
    adjGraph[+to][+from] = true;
  });

  const dfs = (start) => {
    let finished = false;
    let answer = -1;
    const visited = new Set([start]);

    const helper = (cur, picked) => {
      if (finished) return;
      if (picked.length === K) {
        answer = [...picked];
        finished = true;
        return;
      };

      for (let next = cur + 1; next <= N; next++) {
        if (visited.has(next)) continue;

        const isFriend = picked.reduce((acc, pick) => acc && adjGraph[pick][next], true);
        if (isFriend) {
          visited.add(next);
          picked.push(next);
          helper(next, picked);
          picked.pop();
        }
      }
    }

    helper(start, [start]);

    return answer;
  }

  for (let start = 1; start <= N; start++) {
    answer = dfs(start);
    if (answer !== -1) return answer.join("\n");
  }

  return -1;
}

File: BOJ/[2026] 소풍/Johoseong/2026.py
import sys
input = sys.stdin.readline

def dfs(N, cur, arr):
    for i in arr: # 지금까지 친구관계<->현재 노드가 모두 친구 관계인지 확인
        if not friends[cur][i]:
            return [-1]

    if len(arr) + 1 == K: # 친구 관계가 K명 만족
        for i in (arr + [cur]):
            print(i)
        exit()

    for i in range(cur, N + 1):
        if friends[cur][i]:
            dfs(N, i, arr + [cur])

K, N, F = map(int, input().split())
friends = [[0] * (N + 1) for _ in range(N + 1)]
answer = []

for _ in range(F):
    a, b = map(int, input().split())
    friends[a][b] = b
    friends[b][a] = a

for i in range(1, N - K + 2):
    for j in range(i, N + 1):
        if friends[i][j]:
            answer = dfs(N, j, [i])
print(-1)